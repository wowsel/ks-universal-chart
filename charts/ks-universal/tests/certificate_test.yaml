suite: test certificate
templates:
  - deployment.yaml
  - certificate.yaml
tests:
  - it: should create a certificate when autoCreateCertificate is enabled
    set:
      generic:
        ingressesGeneral:
          domain: "example.org"
      deployments:
        test-app:
          replicas: 1
          autoCreateCertificate: true
          autoCreateIngress: true
          containers:
            main:
              image: nginx
              imageTag: latest
              ports:
                http:
                  containerPort: 80
          ingress:
            hosts:
              - subdomain: "test"
                paths:
                  - path: "/"
                    pathType: "Prefix"
              - host: "example.com"
                paths:
                  - path: "/"
                    pathType: "Prefix"
          certificate:
            clusterIssuer: "letsencrypt-prod"
    asserts:
      - matchSnapshot:
          path: spec 