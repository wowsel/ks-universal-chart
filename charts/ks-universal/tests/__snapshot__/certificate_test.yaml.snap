should create a certificate when autoCreateCertificate is enabled:
  1: |
    replicas: 1
    selector:
      matchLabels:
        app.kubernetes.io/component: test-app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: test-app
    template:
      metadata:
        labels:
          app.kubernetes.io/component: test-app
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/managed-by: Helm
          app.kubernetes.io/name: test-app
          helm.sh/chart: ks-universal-test-version
      spec:
        containers:
          - image: nginx:latest
            name: main
            ports:
              - containerPort: 80
                name: http
                protocol: TCP
  2: |
    rules:
      - host: test.example.org
        http:
          paths:
            - backend:
                service:
                  name: test-app
                  port:
                    name: http
              path: /
              pathType: Prefix
      - host: example.com
        http:
          paths:
            - backend:
                service:
                  name: test-app
                  port:
                    name: http
              path: /
              pathType: Prefix
    tls:
      - hosts:
          - test.example.org
          - example.com
        secretName: test-app-tls
  3: |
    dnsNames:
      - test.example.org
      - example.com
    issuerRef:
      kind: ClusterIssuer
      name: letsencrypt-prod
    secretName: test-app-tls
